sdk/identity/azure-identity/azure/identity/_authn_client.py

```diff
@@ -128,14 +128,24 @@ def _deserialize_and_cache_token(self, response, scopes, request_time):
     @staticmethod
     def _parse_app_service_expires_on(expires_on):
         # type: (str) -> struct_time
-        """Parse expires_on from an App Service MSI response (e.g. "06/19/2019 23:42:01 +00:00") to struct_time.
-        Expects the time is given in UTC (i.e. has offset +00:00).
+        """Parse an App Service MSI version 2017-09-01 expires_on value to struct_time.
+
+        This version of the API returns expires_on as a UTC datetime string rather than epoch seconds. The string's
+        format depends on the OS. Responses on Windows include AM/PM, for example "1/16/2020 5:24:12 AM +00:00".
+        Responses on Linux do not, for example "06/20/2019 02:57:58 +00:00".
+
+        :raises ValueError: ``expires_on`` didn't match an expected format
         """
-        if not expires_on.endswith(" +00:00"):
-            raise ValueError("'{}' doesn't match expected format".format(expires_on))
```

Comment:
This is still valid, right?

Reply:
Yes, and the new code raises the same exception for unexpected strings.

Reply:
It seems to me  the time ends with " +08:00" does not raise.

Reply:
That's correct. We assume the endpoint returns time in UTC as documented.